graphql_client_codegen/src/codegen.rs

```diff
@@ -165,7 +165,10 @@ fn generate_scalar_definitions<'a, 'schema: 'a>(
 }
 
 fn render_derives<'a>(derives: impl Iterator<Item = &'a str>) -> impl quote::ToTokens {
-    let idents = derives.map(|s| Ident::new(s, Span::call_site()));
+    let idents = derives.map(|s| {
+        syn::parse_str::<syn::Path>(s)
+            .expect(format!("couldn't parse {} as a derive Path", s).as_str())
```

Comment:
This will allocate a string even in the success case.

This could be changed to something like:

```rust
syn::parse_str::<syn::Path>(s).map_err(|_| format!("couldn't parse {} as a derive Path", s).as_str()).unwrap()
```

Reply:
Oh indeed, thanks a lot for pointing it out!